ops/pebble.py

```diff
@@ -181,6 +182,19 @@
     total=False,
 )
 
+LocalIdentityDict = typing.TypedDict('LocalIdentityDict', {'user-id': int})
+BasicIdentityDict = typing.TypedDict('BasicIdentityDict', {'password': str})
+
+IdentityDict = typing.TypedDict(
+    'IdentityDict',
+    {
+        'access': Literal['untrusted', 'metrics', 'read', 'admin'],
+        'local': Optional[LocalIdentityDict],
+        'basic': Optional[BasicIdentityDict],
+    },
+    total=False,
+)
```

Comment:
This is my suggestion to use `NotRequired`. I think we really do want these values to be `NotRequired` in the `TypedDict`, and I also think we don't want them to be `Optional`.

This change may require an additional ignore in tests (where we intentionally use a bad dict), and should let us drop the `cast` in `from_dict`.

However, I'm not 100% convinced that we have the type of `access` correct here and in `class Identity`, so more to come.

```suggestion

class IdentityDict(typing.TypedDict):
    """TypedDict for Pebble identy."""

    access: Literal['untrusted', 'metrics', 'read', 'admin']
    local: NotRequired[LocalIdentityDict]
    basic: NotRequired[BasicIdentityDict]
```